src/resolvers/mutation.ts

```diff
@@ -385,3 +386,37 @@ export async function updateTag(
     );
   }
 }
+
+/**
+ * Replaces all tags for a given saved item with the tags provided in the input
+ * @param root
+ * @param args
+ * @param context
+ */
+export async function replaceSavedItemTags(
+  root,
+  args: { input: SavedItemTagsInput[] },
+  context: IContext
+): Promise<SavedItem[]> {
+  const savedItems = await new TagDataService(
+    context,
+    new SavedItemDataService(context)
+  ).replaceSavedItemTags(args.input);
+
+  const savedItemTagsMap = args.input.reduce((savedItemTags, input) => {
```

Comment:
having a similar implementation in createSavedItemTags and thinking about this case for emitting snowplow events.

what happens if there are duplicates, e.g
```
[
{savedItemIdA, [{tagA, tagA}]},
]
```

i see that we remove duplicates in insertTags, but we can change tags array as `set` before emitting events